plugin/pkg/shoot/dns/admission.go

```diff
@@ -301,7 +309,15 @@ func assignDefaultDomainIfNeeded(shoot *core.Shoot, projectLister corelisters.Pr
 			if shoot.Spec.DNS == nil {
 				shoot.Spec.DNS = &core.DNS{}
 			}
-			generatedDomain := fmt.Sprintf("%s.%s.%s", shoot.Name, project.Name, domain)
+			shootDNSName := shoot.Name
+			if len(shoot.Name) == 0 && len(shoot.GenerateName) > 0 {
+				random, err := gardenerutils.GenerateRandomStringFromCharset(15, "0123456789abcdefghijklmnopqrstuvwxyz")
+				if err != nil {
+					return apierrors.NewInternalError(err)
+				}
+				shootDNSName = fmt.Sprintf("shoot-%s", random)
```

Comment:
I don't think we need this prefix here. let's just take the random name computed above.